src/core/server/metrics/integration_tests/metrics_service.test.ts

```diff
@@ -0,0 +1,195 @@
+      const testSettings = {
+        ops: {
+          interval: 500,
+        },
+        logging: {
+          silent: true, // set "true" in kbnTestServer
+          appenders: {
+            'custom-console': {
+              kind: 'console',
+              layout: {
+                highlight: false,
+                kind: 'pattern',
+                pattern: '%level|%logger',
+              },
+            },
+          },
+          root: {
+            appenders: ['custom-console', 'default'],
+            level: 'warn',
+          },
+          loggers: [
+            {
+              context: 'metrics.ops',
+              appenders: ['custom-console'],
+              level: 'debug',
+            },
+          ],
+        },
+        plugins: {
+          initialize: false,
+        },
+      };
+      root = kbnTestServer.createRoot({ ...testSettings });
+      coreSetup = await root.setup();
+
+      coreSetup.metrics.getOpsMetrics$().subscribe((opsMetrics) => {
+        testData = opsMetrics;
+      });
+
+      expect(mockConsoleLog).toHaveBeenCalledTimes(1);
+      const [level, logger] = mockConsoleLog.mock.calls[0][0].split('|');
+      expect(level).toBe('DEBUG');
+      expect(logger).toBe('metrics.ops');
+
+      await root.shutdown();
+    });
+  });
+  describe('ops metrics logging content', () => {
+    it('logs memory, uptime, load and delay ops metrics in the message', async () => {
+      root = kbnTestServer.createRoot({ ...otherTestSettings });
+      coreSetup = await root.setup();
+
+      coreSetup.metrics.getOpsMetrics$().subscribe((opsMetrics) => {
+        testData = opsMetrics;
+      });
+      const expectedArray = ['memory:', 'uptime:', 'load:', 'delay:'];
+
+      expect(testData).toBeTruthy();
+      expect(mockConsoleLog).toHaveBeenCalledTimes(1);
+      const [message] = mockConsoleLog.mock.calls[0][0].split('|');
+
+      const messageParts = message.split(' ');
+      const testParts = [messageParts[0], messageParts[2], messageParts[4], messageParts[6]];
+      // the contents of the message are variable based on the process environment,
+      // so we are only performing assertions against parts of the string
+      expect(testParts).toEqual(expect.arrayContaining(expectedArray));
+
+      await root.shutdown();
+    });
+
+    it('logs structured data in the log meta', async () => {
+      root = kbnTestServer.createRoot({ ...otherTestSettings });
+      coreSetup = await root.setup();
+
+      coreSetup.metrics.getOpsMetrics$().subscribe((opsMetrics) => {
+        testData = opsMetrics;
+      });
+      const [, meta] = mockConsoleLog.mock.calls[0][0].split('|');
+      expect(Object.keys(JSON.parse(meta).host.os.load)).toEqual(['1m', '5m', '15m']);
+      expect(Object.keys(JSON.parse(meta).process)).toEqual(expect.arrayContaining(['uptime']));
+
+      await root.shutdown();
+    });
+
+    it('logs ECS fields in the log meta', async () => {
```

Comment:
nit: This probably doesn't need to be an integration test & could be covered by unit tests.

Since you are testing static data returned by `getEcsOpsMetricsLog`, there isn't really an "integration" you are testing against here.

Reply:
You're right! I've moved the test: https://github.com/elastic/kibana/pull/88070/files#diff-dead100406e458541c1c9aee707ebeadc9c8b12de2f5e9970b1048184ae9976fR123